### ISSUE TITLE
heraldEscalte & check

### ISSUE THREAD

**Issue by SachaG** [Meteor-Reaction/Herald#66]
When using the `audit-arguments-check` package, I get the following error:

```
Exception in defer callback: Error: Did not check() all arguments during internal call to 'heraldEscalate'
```

It does seem like `heraldEscalate` does not use `check`: https://github.com/Meteor-Reaction/Herald/blob/d057a2b1f7e92603f89066cd2880f9edf652b86b/server/escalate.js#L35-L41

**Comment by Kestanous** [comment]
Sorry for the very late response. Ah, no it does not. That should be fixed. I will look into it.

**Comment by Kestanous** [comment]
I seems like the ideal fix would be to put all the checks into the server [escalate](https://github.com/Meteor-Reaction/Herald/blob/d057a2b1f7e92603f89066cd2880f9edf652b86b/server/escalate.js#L45) call. This way check will run even if you are calling it from the server.

**Comment by Kestanous** [comment]
It ended up being easier to put it directly in the method. Changes are in current master. When I get the chance, hopefully soon, I will see about merging and closing PRs and publishing an updated Herald.

**Comment by SachaG** [comment]
Awesome, great to see some activity on Herald again :)

**Comment by Kestanous** [comment]
Yeah sorry about that, been really busy with work. And frustratingly we still don't need Herald :/ though we will "at some point". That said I don't want it to die so got to give it some love ^_^

**Comment by jstoeffler** [comment]
Here's some love!
Are you still planning on publishing the fixed version?
I love herald a lot more than audit-arguments-check, but I'm not ready to remove the safety net :)

**Comment by jstoeffler** [comment]
I've just installed the master branch on my project, and I have an error :


    Error: Match error: Expected particular constructor
    //...
    at [object Object].Meteor.methods.heraldEscalate (packages/kestanous:herald/server/escalate.js:36:1)

Which is from

    check(notificationId, Meteor.Collection.ObjectID)

I found this [SO answer](http://stackoverflow.com/a/20850448/183904), which says this method is appropriate, but apparently it's outdated. Did you check() your code? :)

I've changed the code to 

    check(notificationId, String)

and it works now. It's maybe not very strict, but it works, and shouldn't cause any problem.

BTW: missing semicolons after the two checks in this file too.

**Comment by Kestanous** [comment]
Please see #68 very sorry. I don't want to try just pushing since I don't know the current state of things anymore.

**Comment by jstoeffler** [comment]
Ok, it makes sense. If you don't find anyone in the next weeks, let me know.

**Comment by Kestanous** [comment]
Well do, also just in case you did not know. You can clone this to yourApp/packages/ and update the code there. Meteor will pull from there rather instead. This way you can get the current code and patch up anything you need. Sometimes it helps to increment the version number packages.js file to be sure it is using the local package.

**Comment by jstoeffler** [comment]
Thanks, I know :)

It's nice to see how you care about this package. Also it's a bit confusing why there's so few interest in this, it's a very common functionality.

I'm going to use it in production in a couple of days, I'm really not sure if I'll have the time to take care of an open source project (this is why I prefer someone else to do it), but if no one else offers help, let me know. There are so many things I do I don't care about, maybe I could try...

Best

Julien
